src/odemis/gui/cont/tabs.py

```diff
@@ -1602,6 +1611,419 @@ def get_display_priority(cls, main_data):
+            raise ValueError('The stage is missing FAV_POS_DEACTIVE, FAV_POS_ACTIVE and FAV_POS_COATING metadata.')
+        self.target_position_metadata = {LOADING: stage_metadata[model.MD_FAV_POS_DEACTIVE],
+                                         IMAGING: stage_metadata[model.MD_FAV_POS_ACTIVE],
+                                         COATING: stage_metadata[model.MD_FAV_POS_COATING], }
+
+        # Determine and show current position of the stage
+        self.current_position, self.target_position = None, None
+        self._enable_movement_controls()
+        if self.current_position in self.position_btns.keys():
+            pos_button = next(button for pos, button in self.position_btns.items() if pos == self.current_position)
+            self._toggle_switch_buttons(pos_button)
+        self._show_cancel_warning_msg(None)
+
+        # Event binding for tab controls
+        panel.btn_switch_loading.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+        panel.btn_switch_imaging.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+        panel.btn_switch_milling.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+        panel.btn_switch_coating.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+        panel.stage_align_btn_p_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.stage_align_btn_m_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.stage_align_btn_p_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.stage_align_btn_m_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.stage_align_btn_p_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.stage_align_btn_m_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+        panel.btn_switch_advanced.Bind(wx.EVT_BUTTON, self._on_show_advanced)
+        panel.btn_cancel.Bind(wx.EVT_BUTTON, self._on_cancel)
+
+    def _on_change_project_folder(self, evt):
+        """
+        Shows a dialog to change the path and name of the project directory.
+        returns nothing, but updates .conf and project path text control
+        """
+        # Generate suggestion for the new project name to show it on the file dialog
+        np = create_projectname(self.conf.last_path, self.conf.fn_ptn, self.conf.fn_count)
+        new_name = ShowChamberFileDialog(self._tab_panel, np)
+        if new_name is not None:
+            # Create project directory and if it exists warn for existing files
+            if not os.path.isdir(new_name):
+                os.mkdir(new_name)
+            elif os.listdir(new_name):
+                logging.warning("Selected directory {} already contains files.".format(new_name))
+            # If the sample is already loaded, move files that exist in previous folder
+            prev_dir = os.path.join(self.conf.last_path, self.conf.fn_count)
+            if self.current_position != LOADING and os.path.isdir(prev_dir):
```

Comment:
Is this enough check that the sample is already loaded?

Reply:
I'm not convinced by the logic. I don't think it should behave differently if it's in LOADING position. It's only when starting the GUI, or unloading that it should create a new directory, immediately. That's probably a separate (simpler function).
If the previous directory exists (which is most likely the case), use `os.rename()`. If rename fails, I think there are mostly two reasons:
* It's in a different partition => do a complete directory move
* It's inside a the current directory => If there are already files in it, tell the user it's not possible. If the prev_dir is empty, create a directory inside it.
* For all other reasons, log the exception, tell the user it failed, and do not change the folder name.


Also, avoid using the logging infrastructure to indicate an issue to the user. If it goes wrong, it should show an error dialogbox. If it's about to do something really dangerous, ask the user in a dialogbox.